woocommerce/woocommerce
# New CFE workflow - Generate changelog during build step

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

Changelog generation during the build step will help reduce the complexity of changelog management we do while cherry-picking.

In this PR I've created a new workflow, which can eventually be combined as part of the build step with further refinement. 







### How to test the changes in this Pull Request:



I am sure there must be a better way to test this, but these are the steps I followed:

Please disable some common workflows in your forked repo, like CI etc to save time.

1. Fork this repo in a different org, make sure the name is `woocommerce`.
2. Add this as a remote in your local.
3. Create a new branch `release/9.4` out of this branch and push that upstream in your new repo.
4. Goto actions, and check if the workflow is present, if not change the default branch to `release.9.3` then trigger the workflow, then switch it back to `trunk` immediately from the `settings > general > default branch`.
5. Now, trigger this workflow manually to create a changelog PR for `release/9.4` branch in actions tab.
6. You should receive two PR links in the end, one like [this](https://github.com/naman03malhotra/woocommerce/pull/10) that modifies the changelog for the release branch and another one like [this](https://github.com/naman03malhotra/woocommerce/pull/11) that will delete the changelog from `trunk`

Note: This workflow will be manually triggered during the build package day, eventually we can merge it with the build flow once it is stable. 


<img width="961" alt="image" src="https://github.com/user-attachments/assets/7ffe2c8e-e46d-4214-8a6b-e6eb310f80e9">
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/95720d9f-13a5-41dd-bdbf-40af07f16d6d">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
